jax/scipy/ndimage.py

```diff
@@ -100,7 +100,9 @@ def _map_coordinates(input, coordinates, order, mode, cval):
       contribution = jnp.where(all_valid, input[indices], cval)
     outputs.append(_nonempty_prod(weights) * contribution)
   result = _nonempty_sum(outputs)
-  return result
+  if jnp.issubdtype(input.dtype, jnp.integer):
+    result = jnp.around(result)
+  return result.astype(input.dtype)
```

Comment:
This behavior for integer valued dtypes is not what I would have guessed for non-nearest neighbor interpolation! But I guess we don’t have any good reason to deviate from SciPy’s behavior here.

Reply:
Ok -- you're right to be skeptical. Round half up matches scipy here -- need to flesh out the test again to catch this case as well.

Reply:
fixed